llvm/lib/Transforms/IPO/DynUntrustedAllocPre.cpp

```diff
@@ -251,13 +261,17 @@ class DynUntrustedAllocPre : public ModulePass {
     auto rust_dealloc = M.getFunction("__rust_dealloc");
 
     auto rust_untrusted_alloc = M.getFunction("__rust_untrusted_alloc");
+    assert(rust_untrusted_alloc != nullptr && "Missing rust_untrusted_alloc.");
     auto rust_untrusted_alloc_zeroed =
         M.getFunction("__rust_untrusted_alloc_zeroed");
+    //assert(rust_untrusted_alloc_zeroed != nullptr && "Missing rust_untrusted_alloc_zeroed.");
```

Comment:
why not assert on this? if its because you think it may not be used/called, there is no guarantee that code just doesn't exclusively use this, and never calls `__rust_alloc()`.

For now, we can use some simplifying assumptions for eval, but we should address this

Reply:
I dont think we can assert on this as it is possible that this never gets called either. We may want to move away from hard asserts on either one and just add conditional checks for those functions existing.